### ISSUE TITLE
Partial Refund issue

### ISSUE THREAD

**Issue by tajriba** [MultiSafepay/prestashop-official#4]
Latest Version: 5.6.
Bug: When refund an order, the total order amount is being refunded.
Example: Push partial refund. And total order amount refund is send to MSP.

When you have 2 products in the order, only the partial refund (amount of 1 of the 2 products) need to be refunded.
<img width="742" alt="Scherm­afbeelding 2022-12-28 om 18 07 01" src="https://user-images.githubusercontent.com/68469531/209847497-1b3b451e-09f1-438c-abb5-30cbd861e207.png">

<img width="547" alt="afbeelding" src="https://user-images.githubusercontent.com/68469531/209847438-8df6070b-5ecc-458a-8d14-758b6670e87b.png">

**Comment by danielcivit** [comment]
Hello @tajriba.

Thanks for contact us and report this issue.  Sadly, for the moment I am not able to reproduce this issue. The procedure I am following is the next one: 

1. Order with payment accepted. 
2. Browse to Orders > Orders > Select the order > Partial Refund. 
3. Set the items you want to refund. As an example, the following screenshot in which only one item is being selected to be refunded The first one. 

<img width="1224" alt="Screenshot 2022-12-29 at 10 26 01" src="https://user-images.githubusercontent.com/64795062/209931706-0f3a9d35-a2f6-4593-b388-0eb7c3080506.png">

4. Refund is processed properly, and refers only to that item and that amount. 

<img width="735" alt="Screenshot 2022-12-29 at 10 27 34" src="https://user-images.githubusercontent.com/64795062/209931755-66eab644-c751-477c-9ed6-8f325219b1a7.png">

Maybe there was a mistake in this procedure. You enter the wrong input. Or something else related with differences in our environment or settings.

Could you try again? 

Maybe in a test - staging environment, using your test account, and confirm this is happening and it was not related with an error in the procedure, at the moment you processed the partial refund.

**Comment by tajriba** [comment]
Beste,

In verband met vakantie zijn wij niet bereikbaar van 24 december t/m 1 januari. Betreft het een dringend spoedgeval, dan zijn wij per mail bereikbaar op ***@***.***

Met vriendelijke groet,
Stefan | Tajriba

**Comment by tajriba** [comment]
Hi,
I have tested this in 2 real environments. PrestaShop version 1.7.6.X.
I see you have tested it with the newer version?

**Comment by danielcivit** [comment]
Hi @tajriba 

That is correct. I`m going to test this using 1.7.6.X and get back to you with more information. 

Thanks again for the report.

**Comment by danielcivit** [comment]
Hi @tajriba.

We was able to reproduce this issue and confirm this was affecting PrestaShop versions 1.7.6.X. 

We just released a new version, which is already available in GitHub or from our docs website.  Upgrade instruction can also be found in docs website. 

https://github.com/MultiSafepay/prestashop-official/releases/tag/5.6.1
https://docs.multisafepay.com/docs/prestashop-1-7

Once again, thanks for your report.

**Comment by tajriba** [comment]
@Daniel-MultiSafepay We have found another issue related to this. This is happening for a long time.

When you do Partial refund and select the option: Generate gift card. The order amount needs not the be sended to MSP.
Because a gift card as refund amount is generated for the customer. No refund payment has been done.

This is now happening, so the customer gets a gift card an get his money back...

<img width="964" alt="Scherm­afbeelding 2023-01-10 om 14 43 11" src="https://user-images.githubusercontent.com/68469531/211568114-dda971ef-746d-4c8e-bb1b-5ab3d86b1e34.png">

**Comment by danielcivit** [comment]
Hi @tajriba.

Thanks for your report. 

I`m going to immediately check this issue and get back to you with more information as soon as possible.

**Comment by danielcivit** [comment]
Hi @tajriba.

We was able to reproduce this issue and confirm this was affecting PrestaShop versions 1.7.6.X.

Currently working in a new version where this issue will be fixed. I will get back to you once is done. 

Thanks again for your report.

**Comment by danielcivit** [comment]
Hi @tajriba 

We just released a new version (5.6.2), which is already available in GitHub or from our docs website. Upgrade instruction can also be found in docs website.

- https://github.com/MultiSafepay/prestashop-official/releases/tag/5.6.2
- https://docs.multisafepay.com/docs/prestashop-1-7

Once again, thanks for your report about these issues. 

For the moment I am going to close this ticket, but don't hesitate to get back to us if you find anything else, questions, or feedback

**Comment by tajriba** [comment]
@Daniel-MultiSafepay Can you also check for the shipping costs?
Our client reports this: If have setup a refund in the backoffice of prestashop and filled the box to also refund the shipping but the amount is send as refund to MSP without shipping costs.
<img width="560" alt="Scherm­afbeelding 2023-01-25 om 10 53 27" src="https://user-images.githubusercontent.com/68469531/214532532-b5c100c2-1a70-4b25-97f2-5b563b558b8d.png">

 (also on version lower than 1.7.7.0) his shop will be soon updated to the latest version. I have not tested this on newer versions.

**Comment by danielcivit** [comment]
Hi @tajriba.

Once again, thanks for contact us.

I will be checking this report, and get back to you as soon as possible with more information.

**Comment by danielcivit** [comment]
Hi @tajriba.

We was able to reproduce the latest reported issue, and confirm this was affecting PrestaShop versions 1.7.6.X.

We just released a new version (v. 5.7.0), which is already available in GitHub or from our docs website. Upgrade instruction can also be found in docs website.

https://github.com/MultiSafepay/prestashop-official/releases/tag/5.7.0
https://docs.multisafepay.com/docs/prestashop-1-7

Hopefully, after all your reports, partial refunds will work properly now in versions 1.7.6.X

Once again, thanks for contact us and don't hesitate to get back if you found any other issue.
